### PR TITLE
[test] test whether the RCP support CSL transmitter

### PR DESCRIPTION
**Pull request by zhanglongxia**
This commit adds a test case to `cp-tests` to test whether the RCP supports the CSL transmitter.

### CODE REVIEW

**Review by superwhd — APPROVED** [review]
LGTM overall

File: tools/cp-caps/rcp_caps_test.py [hunk 1]
```diff
@@ -79,9 +79,45 @@ def test_diag_commands(self):
...
+        # Set the reference device as a SSED
+        self.__ref.set_mode('-')
+        self.__ref.config_csl(channel=15, period=320000, timeout=100)
+        self.__ref.join(self.__dataset)
+        self.__ref.wait_for('state', 'child')
+
+        child_table = self.__dut.get_child_table()
+        ret = len(child_table) == 1 and child_table[1]['csl'] is True
+
+        if ret is True:
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:110]:
```suggestion
        if ret:
```

File: tools/otci/otci/otci.py [hunk 2]
```diff
@@ -746,7 +746,8 @@ def get_child_table(self) -> Dict[ChildId, Dict[str, Any]]:
 
             id = int(col("ID"))
             r, d, n = int(col("R")), int(col("D")), int(col("N"))
-            mode = DeviceMode(f'{"r" if r else ""}{"d" if d else ""}{"n" if n else ""}')
+            mode = DeviceMode(
+                f'{"r" if r else ""}{"d" if d else ""}{"n" if n else ""}{"-" if r == 0 and d == 0 and n == 0 else ""}')
```
Comment (superwhd) [inline tools/otci/otci/otci.py:750]: It may be worth a helper function for this

**Review by Irving-cl — APPROVED** [review]
LGTM 👍

File: tools/cp-caps/rcp_caps_test.py [hunk 3]
```diff
@@ -79,9 +79,44 @@ def test_diag_commands(self):
...
+        self.__ref.join(self.__dataset)
+        self.__ref.wait_for('state', 'child')
+
+        child_table = self.__dut.get_child_table()
+        ret = len(child_table) == 1 and child_table[1]['csl']
+
+        if ret:
+            ref_mleid = self.__ref.get_ipaddr_mleid()
+            result = self.__dut.ping(ref_mleid, count=packets, interval=1)
+            ret = result['transmitted_packets'] == result['received_packets'] == packets
```
Comment (abtink) [inline tools/cp-caps/rcp_caps_test.py:113]: Nice! I did not know we can do multiple `==` to check if all variables are same in python. 😃

**Review by abtink — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**